compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -543,8 +543,14 @@ object Parsers {
       }
 
     /** Accept identifier and return Ident with its name as a term name. */
-    def termIdent(): Ident = atPos(in.offset) {
-      makeIdent(in.token, ident())
+    def termIdent(): Ident = {
+      val lastOffset = in.lastOffset
+      val id = atPos(in.offset) {
+        makeIdent(in.token, ident())
+      }
+      // Make sure that even trees with parsing errors have a offset that is within the offset
+      if (id.name == nme.ERROR && id.pos == NoPosition) atPos(lastOffset - 1)(id)
+      else id
```

Comment:
Why nothing for `typeIdent`? Also, I think we should put the whole logic in `makeIdent`, like this:
```scala
    private def makeIdent(tok: Token, name: Name) = {
      val tree =
        if (tok == BACKQUOTED_IDENT) BackquotedIdent(name)
        else Ident(name)
      if (name.toTermName == nme.ERROR) tree.withSpan(Span(in.offset, in.offset))
      else tree
    }
```

Reply:
And,..., let's get new source positions in first, otherwise we just get needless rebase breakages.

Reply:
Yes, we should wait until the changes in positions.

Reply:
This change has been implemented.